analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/Invocation.cs

```diff
@@ -196,6 +199,25 @@ private static ProgramState[] ProcessEquals(SymbolicContext context, IInvocation
         return context.State.ToArray();
     }
 
+    private static ProgramState[] ProcessNullableGetValueOrDefault(SymbolicContext context, IInvocationOperationWrapper invocation)
+    {
+        var instanceValue = context.State[invocation.Instance];
+        if (instanceValue?.HasConstraint(ObjectConstraint.Null) is true)
+        {
+            return ConstantCheck.ConstraintFromType(((INamedTypeSymbol)invocation.Instance.Type).TypeArguments.Single()) is { } orDefaultConstraint
```

Comment:
This is a bit cryptic. A code comment like `// Default case: Learn constraints for default values like BoolContraint.False` would be helpful here.

Reply:
I tried to extract `valueType` like this instead. Is it better?
```
var valueType = ((INamedTypeSymbol)invocation.Instance.Type).TypeArguments.Single();
return ConstantCheck.ConstraintFromType(valueType) is { } orDefaultConstraint
```

Reply:
The `valueType` means both, the `.Value type` as well as non-nullable value type :)

Reply:
`var valueType = ((INamedTypeSymbol)invocation.Instance.Type).TypeArguments.Single();` This should an extension method `UnwrapNullable` or something like that.

Reply:
I haven't found other place where it could be used except `IsNullableBoolean`. We can do it later if we have 3rd place